### ISSUE TITLE
Choco new silentargs can't pass in args in the param=value format

### ISSUE THREAD

**Issue by KosherSalt** [chocolatey/choco#510]
Some applications like Operations Manager use a similar format to choco new to do pass in their arguments, e.g. USE_SETTINGS_FROM_AD={0|1} . (https://technet.microsoft.com/en-us/library/hh230736.aspx)

Creating a new package from template with silentarg="/qn /norestart USE_SETTINGS_FROM_AD=0" will create an emtpy string in the chocolateyInstall.ps1 file. I've tried quoting it several different ways (double quotes, single quotes, double quotes around single quotes, etc).
```powershell
silentArgs    = ""
```

This is using Chocolatey v0.9.10-beta1

**Comment by ferventcoder** [comment]
Interesting.

This is in the template? What exactly does it look like?

Like this?
~~~powershell
silentarg="/qn /norestart USE_SETTINGS_FROM_AD=0"
~~~

**Comment by ferventcoder** [comment]
and how are you calling choco new?

**Comment by ferventcoder** [comment]
Ooooh, wait - `choco new silentargs=/qn /norestart USE_SETTINGS_FROM_AD=0"`?

**Comment by ferventcoder** [comment]
To pass arguments with spaces - https://github.com/chocolatey/choco/wiki/CommandsReference#how-to-pass-options--switches

tl;dr: 

Depending on whether you are running the command from powershell.exe or cmd.exe, this is how you should pass it:
`choco new silentargs="'/qn /norestart USE_SETTINGS_FROM_AD=0'"`

**Comment by ferventcoder** [comment]
That is double quotes surrounding apostrophes. Shells make things harder than they need to be sometimes. :/

**Comment by ferventcoder** [comment]
I think you also brought to light that in the default template this is broken - if you create a template and drop it into a template location as a replacement - add the SilentArgs back in as 

`[[SilentArgs]]` - that's what choco uses for template replacement.

**Comment by KosherSalt** [comment]
Running that command with the apostrophes in double quotes doesn't succeed in either cmd or powershell.
```
choco new scom --template embedded silentargs="'/qn /norestart USE_SETTINGS_FROM_AD=0'"
```
In my template, I had added the [[SilentArgs]]:
```powershell
$packageArgs = @{
  packageName   = $packageName
  file          = $fileLocation
  fileType      = '[[InstallerType]]' #only one of these: exe, msi, msu

  silentArgs    = "[[SilentArgs]]"
  validExitCodes= @(0, 3010, 1641)
}
```

Result
```powershell
$packageArgs = @{
  packageName   = $packageName
  file          = $fileLocation
  fileType      = 'EXE_MSI_OR_MSU' #only one of these: exe, msi, msu

  silentArgs    = ""
  validExitCodes= @(0, 3010, 1641)
}
```

This template succeeds if I don't pass the = operator.
```
choco new scom --template embedded silentargs="/qn /norestart USE_SETTINGS_FROM_AD 0"
```

```powershell
$packageArgs = @{
  packageName   = $packageName
  file          = $fileLocation
  fileType      = 'EXE_MSI_OR_MSU' #only one of these: exe, msi, msu

  silentArgs    = "/qn /norestart USE_SETTINGS_FROM_AD 0"
  validExitCodes= @(0, 3010, 1641)
}
```

**Comment by ferventcoder** [comment]
I'm going to try with `-d --noop` specified. That is a good way to see what it is actually assigning where.

**Comment by ferventcoder** [comment]
It's a bug.

**Comment by ferventcoder** [comment]
The code splits on `=`, so that will definitely need to get corrected. Thanks for filing an issue!

**Comment by ferventcoder** [comment]
Do you have a workaround for now?

**Comment by KosherSalt** [comment]
Yeah, we have a workaround. All I have to do is manually edit the file before packing it.
It breaks completely automating the package creation process , but I'm sure I can make something work in powershell to do the same thing until it gets resolved.

**Comment by KosherSalt** [comment]
Thanks for looking into it! I was sure I was going crazy there for a while.

**Comment by ferventcoder** [comment]
Almost fully automated package creation, you will have to tell me about that sometime :+1:

**Comment by sahebjade** [comment]
I have created Pull Request #513 to fix the issue. if you agree with the fix please merge. Thanks
https://travis-ci.org/chocolatey/choco/builds/98298512

**Comment by ferventcoder** [comment]
The missing SilentArgs template value has been missing since 0.9.9.7 - 7b468c8e6ff41913c5fa29499d2ad8cb681a96cb was the commit that introduced the regression.

**Comment by ferventcoder** [comment]
I've added the fix for allowing silentargs again in the default template. I see @sahebjade has created a PR to fix the `=` issue. I've added comments to the PR, if you wouldn't mind looking at those and making adjustments. Thanks!

**Comment by ferventcoder** [comment]
Thanks! I just rebased this to stable and updated the commit message. This will go out in the next beta release in a couple of hours.
